src/snowflake/sqlalchemy/snowdialect.py

```diff
@@ -637,6 +637,9 @@ def _get_table_columns(self, connection, table_name, schema=None, **kw):
 
             type_instance = col_type(**col_type_kw)
 
+            if table_name not in schema_primary_keys:
+                raise sa_exc.NoSuchTableError()
```

Comment:
may I ask why check is added here? the following line:
`current_table_pks = schema_primary_keys.get(table_name)` returns `None` if table not in `schema_primary_keys`.

also the added test doesn't hit the code path I think, do you have a scenario for it?

Reply:
You're correct, but I think it would still cause an issue with L655, where we use the indexing operator on the same dictionary looking for the primary keys, which would throw a KeyError. So I think we still need it. I can add a comment on the check to clarify why we have it if that makes more sense?

Reply:
On your second question, we do not test it, but I think it would show up in the same use case as the other check, but only if there were too many columns for `_get_schema_columns` to return all of them. Then we fall into this function call on L674 and would hit this KeyError while returning, skipping the check on L676.

Reply:
L655 `in schema_primary_keys[table_name]["constrained_columns"]` is only executed when L657 `if current_table_pks` is not None/False and `current_table_pks 
 = schema_primary_keys.get(table_name)` on L643. so L655 shouldn't be executed if `table_name` not in schema_primary_keys.

with the change, it actually changes the behavior as previously this `_get_table_columns` can return if table_name is None but now its raising Error.

Reply:
Ah OK, I see now. Thanks for explaining this.

I'm trying to replicate this code path to make sure the original bug and my original use case will still be fixed if we revert this piece. I've created 10,000 tables with 10,000 columns each in a schema, but it still doesn't seem to hit this code path, and I can't find any documentation online of the error 90030, which would lead to this code path. How would you suggest I reproduce this code path for testing purposes in my own account?

Thanks!

Reply:
I think I know why -- it's because this check is in for-looping the result, and when there is no table in the schema, the result is empty so the code path doesn't get hit as the for-loop doesn't get run at all.

To align the error experience, one thing we could do is to check whether `ans` is empty in the end -- if it's empty, then it means no table found in the schema. Do you want to try it on your end?

Reply:
@sfc-gh-aling - Thanks for this, I think that approach makes sense. I've gone ahead and made it, so please take a look and verify it is what you were expecting.

Reply:
change looks good to me, thanks!